### ISSUE TITLE
Index value is always -1

### ISSUE THREAD

**Issue by CyberAndrii** [axuno/SmartFormat#134]
Always returns `-1` but should be `2`:
```cs
Smart.Format("{Index}", new { Index = 2 });
```

**Comment by axunonb** [comment]
Assuming you're using the default setting, the ```ListFormatter``` will be used as an ```ISource``` and ```IFormatter``` in your example. Here "Index" as a property name is taken to access objects of a list, which is not what you want.
So you can e.g. either choose another name instead of "Index" or remove the ```ListFormatter```:
```
Smart.Default.FormatterExtensions.Remove(Smart.Default.GetFormatterExtension<ListFormatter>());
Smart.Default.SourceExtensions.Remove(Smart.Default.GetSourceExtension<ListFormatter>());
```

**Comment by axunonb** [comment]
This commit should resolve the problem: https://github.com/axuno/SmartFormat/commit/d3130ab22a07344c26b9de66aa1b3c1ba4598768

**Comment by axunonb** [comment]
Published [2.5.3-preview-1](https://github.com/axuno/SmartFormat/releases/tag/v2.5.3.0-preview1)

**Comment by CyberAndrii** [comment]
Thanks. It is fixed now.
